tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,647 @@
+"""
+This file compares DictConfig methods with the corresponding
+methods of standard python's dict.
+The following methods are compared:
+    __contains__
+    __delitem__
+    __eq__
+    __getitem__
+    __setitem__
+    get
+    pop
+    keys
+    values
+    items
+
+We have separate test classes for the following cases:
+    TestUntypedDictConfig: for DictConfig without a set key_type
+    TestPrimitiveTypeDunderMethods: for DictConfig where key_type is primitive
+    TestEnumTypeDunderMethods: for DictConfig where key_type is Enum
+"""
+from copy import deepcopy
+from enum import Enum
+from typing import Any, Dict, Optional
+
+from pytest import fixture, mark, param, raises
+
+from omegaconf import DictConfig, OmegaConf
+from omegaconf.errors import ConfigKeyError, ConfigTypeError, KeyValidationError
+from tests import Enum1
+
+
+@fixture(
+    params=[
+        "str",
+        1,
+        3.1415,
+        True,
+        Enum1.FOO,
+    ]
+)
+def key(request: Any) -> Any:
+    """A key to test indexing into DictConfig."""
+    return request.param
+
+
+@fixture
+def python_dict(data: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Just a standard python dictionary, to be used in comparison with DictConfig."""
+    return deepcopy(data)
+
+
+@fixture(params=[True, False, None])
+def struct_mode(request: Any) -> Optional[bool]:
+    struct_mode: Optional[bool] = request.param
+    return struct_mode
+
+
+@mark.parametrize(
+    "data",
+    [
+        param({"a": 10}, id="str"),
+        param({1: "a"}, id="int"),
+        param({123.45: "a"}, id="float"),
+        param({True: "a"}, id="bool"),
+        param({Enum1.FOO: "foo"}, id="Enum1"),
```

Comment:
Tests here seems to be focusing on the type of the key only.
What is the purpose of this test? why is the type of the key so critical?